app/Image/FileDeleter.php

```diff
@@ -114,7 +114,9 @@ public function do(): void
 					// The latter part deletes (regular) files, but avoids errors
 					// in case the file doesn't exist.
 					if (is_link($absolutePath) || file_exists($absolutePath)) {
-						if (!unlink($absolutePath)) {
+						try {
+							\Safe\unlink($absolutePath);
+						} catch (\Throwable) {
 							$firstException = $firstException ?: new \RuntimeException('unlink failed: ' . $absolutePath);
```

Comment:
```suggestion
						try {
							\Safe\unlink($absolutePath);
						} catch (\Throwable $e) {
							$firstException = $firstException ?: $e;
```

No need to create a new exception anymore, if `unlink` already provides one.

Reply:
Fixed it myself in https://github.com/LycheeOrg/Lychee/pull/1336/commits/7e9a9ec414515da7c30bc166b412c7bbdf4ccfe7.